src/feat/pitch-functions.cc

```diff
@@ -774,12 +774,17 @@ int32 OnlinePitchFeatureImpl::NumFramesAvailable(
   if (!input_finished_)
     frame_length += nccf_last_lag_;
   if (num_downsampled_samples < frame_length) return 0;
-  else
+  else {
     if (input_finished_ && !snip_edges) {
       return (int32)(num_downsampled_samples * 1.0f / frame_shift + 0.5f);
     }
+    else if (!snip_edges) {
+      return (int32)(
+        (num_downsampled_samples - frame_length/2) * 1.0f / frame_shift + 0.5f);
+    }
     else
```

Comment:
braces much match so put braces on else.  Also it should be
if {
   x;
} else {
  y;
}
i.e. no line break before else.  check rest of your code.